portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -64,6 +68,21 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the request parameters.
+    String originalText = request.getParameter("text");
+    String languageCode = request.getParameter("languageCode");
+
+    // Do the translation.
+    Translate translate = TranslateOptions.getDefaultInstance().getService();
+    Translation translation =
+        translate.translate(originalText, Translate.TranslateOption.targetLanguage(languageCode));
+    String translatedText = translation.getTranslatedText();
+
+    // Output the translation.
+    response.setContentType("text/html; charset=UTF-8");
```

Comment:
consider defining content type as constant. Use `private static final String`

Reply:
Same for character encoding below

Reply:
On an aside note you have added character encoding in both contentType and setCharacterEncoding. Is that expected?